rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,162 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
 
 // ImagePointTo3DPoint takes the 2D pixel point and assumes that it represents the X,Y coordinate in mm as well.
 func (pp *ParallelProjection) ImagePointTo3DPoint(pt image.Point, d rimage.Depth) (r3.Vector, error) {
-	return r3.Vector{float64(pt.X), float64(pt.Y), float64(d)}, nil
+	return r3.Vector{X: float64(pt.X), Y: float64(pt.Y), Z: float64(d)}, nil
+}
+
+// ParallelProjectionOntoXZWithRobotMarker allows the creation of a 2D projection of a pointcloud and robot
+// position onto the XZ plane.
+type ParallelProjectionOntoXZWithRobotMarker struct {
+	robotPose *spatialmath.Pose
+}
+
+const (
+	sigmaLevel       = 7
+	imageHeight      = 1080
+	imageWidth       = 1070
+	missThreshold    = 0.44
+	hitThreshold     = 0.53
+	voxelRadius      = 1
```

Comment:
Is this really a voxel if we are only concerning ourselves with the XZ plane?

Reply:
sigh, you're right. I was wondering if anyone would catch this after we shifted from box to circle. I just like the word voxel T.T

Reply:
:) I like it to but I recommend naming it what it really is to avoid confusion.